Sources/CodableKit/Reflection/Codable+Reflectable.swift

```diff
@@ -50,10 +52,9 @@ extension Decodable where Self: Encodable {
         if let cached = ReflectedPropertyCache.storage[keyPath] {
             return cached
         }
-        
-        var lo = try Self(from: HiLoDecoder(signal: .lo))
+        var lo = try Self(from: HiLoDecoder(signal: .lo, stopOn: keyPath.valueType))
         lo[keyPath: keyPath] = T.reflectDecoded().1
-        let e = HiLoEncoder()
+        let e = HiLoEncoder(stopOn: keyPath.valueType)
```

Comment:
I think we should be able to use `HiLoEncoder<Value>` here or `HiLoEncoder(stopOn: Value.self)`